### PR TITLE
Refactors options to use Enum with fallback for strings

### PR DESCRIPTION
**Pull request by StephenNneji**
Uses a customEnum class so that string values can be different from the enum name

### CODE REVIEW

File: tests/testBackgroundsClass.m [hunk 1]
```diff
@@ -78,23 +78,23 @@ function testAddBackground(testCase)
...
+         testCase.background.addBackground('background 5', allowedTypes.Function, 'function_name', 1, 3);
          testCase.verifyEqual(string(testCase.background.backgrounds.typesTable{end, :}),...
-                              ["background 5", "function", "function_name", "background par 1", "background par 3", "", ""], ...
+                              ["background 5", string(allowedTypes.Function.value), "function_name", "background par 1", "background par 3", "", ""], ...
                               'addBackground method not working');
          testCase.verifyError(@() testCase.background.addBackground('New', 'fixed'), invalidOption.errorID);
-         testCase.verifyError(@() testCase.background.addBackground('New', 'constant'), invalidNumberOfInputs.errorID);
-         testCase.verifyError(@() testCase.background.addBackground('New', 'constant', 6), indexOutOfRange.errorID);
+         testCase.verifyError(@() testCase.background.addBackground('New', allowedTypes.Constant), invalidNumberOfInputs.errorID);
+         testCase.verifyError(@() testCase.background.addBackground('New', allowedTypes.Constant.value, 6), indexOutOfRange.errorID);
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testBackgroundsClass.m:97]: No I am testing the function should accept string i.e value or enum

File: tests/testUtilities.m [hunk 2]
```diff
@@ -8,5 +8,31 @@ function testValidateNumber(testCase)
             testCase.verifyError(@() validateNumber('a'), invalidType.errorID);
             testCase.verifyError(@() validateNumber(false), invalidType.errorID);
         end
+
+        function testValidateOption(testCase)
+            testCase.verifyEqual(validateOption('UniForm', 'priorTypes'), priorTypes.Uniform, ...
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testUtilities.m:13]: I used the case insensitive string comparison so as to keep the backward compatibility. So it should pass as long as the word is spelt correctly but I could take it out later

File: API/projectClass/parametersClass.m [hunk 3]
```diff
@@ -392,7 +386,7 @@ function appendNewRow(obj, row)
             % Appends a new row to the table. Expects a cell array  
             % with row values to append
             % 
-            % obj.appendNewRow({'Tails', 10, 20, 30, true, 'uniform', 0, Inf})
+            % obj.appendNewRow({'Tails', 10, 20, 30, true, 'uniform' 0, Inf})
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/parametersClass.m:389]: I did a replace all that removed the 'uniform' so I probably deleted the comma when I manually reverted

File: tests/testBackgroundsClass.m [hunk 4]
```diff
@@ -130,10 +130,10 @@ function testSetBackground(testCase)
          testCase.background.setBackground(1, 'name', 'Background 1');
          testCase.verifyEqual(testCase.background.backgrounds.typesTable{1, 1}, "Background 1", 'setBackground method not working');
 
-         testCase.background.setBackground(1, 'type', 'constant');
-         testCase.verifyEqual(testCase.background.backgrounds.typesTable{1, 2}, "constant", 'setBackground method not working');
-         testCase.background.setBackground('Background 1', 'type', 'function'); 
-         testCase.verifyEqual(testCase.background.backgrounds.typesTable{1, 2}, "function", 'setBackground method not working');
+         testCase.background.setBackground(1, 'type', 'Constant');
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testBackgroundsClass.m:133]: yes will fix

File: tests/testControlsClass.m [hunk 5]
```diff
@@ -317,7 +317,7 @@ function testDisplayGroups(testCase)
             testCase.verifyThat(display, ~ContainsSubstring("method"), 'getPropertyGroups method not working');
             testCase.verifyThat(display, ~ContainsSubstring("strategy"), 'getPropertyGroups method not working');
             
-            testCase.controls.procedure = 'de';
+            testCase.controls.procedure = 'DE';
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testControlsClass.m:320]: yes will fix

File: tests/testResolutionsClass.m [hunk 6]
```diff
@@ -117,10 +117,10 @@ function testSetResolution(testCase)
             testCase.resolution.setResolution(1, 'name', 'Resolution 1');
             testCase.verifyEqual(testCase.resolution.resolutions.typesTable{1, 1}, "Resolution 1", 'setResolutionValue method not working');
  
-            testCase.resolution.setResolution(1, 'type', 'constant');
-            testCase.verifyEqual(testCase.resolution.resolutions.typesTable{1, 2}, "constant", 'setResolutionValue method not working');
-            testCase.resolution.setResolution('Resolution 1', 'type', 'function'); 
-            testCase.verifyEqual(testCase.resolution.resolutions.typesTable{1, 2}, "function", 'setResolutionValue method not working');
+            testCase.resolution.setResolution(1, 'type', 'Constant');
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply tests/testResolutionsClass.m:120]: yes will fix

File: API/projectClass/layersClassRealSLD.m [hunk 7]
```diff
@@ -54,27 +57,23 @@
...
                 % Add an empty named layer
-                newRow = {layerDetails{1},'','','','','bulk out'};
+                newRow = {layerDetails{1},'','','','',hydrationTypes.BulkOut.value};
             
             else
                 % Add a layer that is fully defined
                 if length(layerDetails) == 4
                     % No hydration
-                    layerDetails = {layerDetails{1},layerDetails{2},layerDetails{3},layerDetails{4},NaN,'bulk in'};
+                    layerDetails = {layerDetails{1},layerDetails{2},layerDetails{3},layerDetails{4},NaN,hydrationTypes.BulkIn.value};
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/layersClassRealSLD.m:70]: I have no idea its must likely a typo

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply API/projectClass/layersClassRealSLD.m:70]: Yeah, that's what I thought.

**Pull request merged**